projects/nlp/junit_test1.py

```diff
@@ -0,0 +1,106 @@
+#!/usr/bin/env python
+# ----------------------------------------------------------------------
+# Numenta Platform for Intelligent Computing (NuPIC)
+# Copyright (C) 2016, Numenta, Inc.  Unless you have purchased from
+# Numenta, Inc. a separate commercial license for this software code, the
+# following terms and conditions apply:
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+# See the GNU Affero Public License for more details.
+#
+# You should have received a copy of the GNU Affero Public License
+# along with this program.  If not, see http://www.gnu.org/licenses.
+#
+# http://numenta.org/licenses/
+# ----------------------------------------------------------------------
+
+helpStr = """
+  Simple script to run unit test 1:
+
+  The dataset is categories defined by base sentences (each with ten words).
+  For each base sentence there are five new sentences each, with an additional
+  word substitution that does not change the meaning of the sentence. For the
+  test we use each of the sentences as a search term. A perfect result ranks the
+  four similar sentences closest to the search.
+"""
+
+import argparse
+
+from projects.nlp.junit_testing import (
+  printResults, setupExperiment, testModel)
+
+
+
+def runExperiment(args):
+  """ Build a model and test it."""
+  model, dataSet = setupExperiment(args)
+
+  _, ranks = testModel(model,
+                       dataSet,
+                       args.verbosity)
+  printResults("JUnit1", ranks)
+
+  return model
+
+
+
+if __name__ == "__main__":
+
+  parser = argparse.ArgumentParser(
+    formatter_class=argparse.ArgumentDefaultsHelpFormatter,
+    description=helpStr
+  )
+
+  parser.add_argument("-c", "--networkConfigPath",
+                      default="data/network_configs/sensor_knn.json",
+                      help="Path to JSON specifying the network params.",
+                      type=str)
+  parser.add_argument("-m", "--modelName",
+                      default="htm",
+                      type=str,
+                      help="Name of model class. Options: [keywords,htm]")
+  parser.add_argument("--retinaScaling",
+                      default=1.0,
+                      type=float,
+                      help="Factor by which to scale the Cortical.io retina.")
+  parser.add_argument("--maxSparsity",
+                      default=1.0,
+                      type=float,
+                      help="Maximum sparsity of Cio encodings.")
+  parser.add_argument("--numLabels",
+                      default=6,
```

Comment:
The default for numLabels should be specific to the unit test - it should include all the categories.